gitsign: 0.4.1 -> 0.6.0

LeSuisse opened the pull request:
###### Description of changes

https://github.com/sigstore/gitsign/releases/tag/v0.5.2
https://github.com/sigstore/gitsign/releases/tag/v0.6.0
<!--
For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->
Result of `nixpkgs-review` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>gitsign</li>
  </ul>
</details>

mweinelt commented:
Builds on aarch64-linux.

Failing test on aarch64-darwin:
```
gitsign-aarch64-darwin> running tests
gitsign-aarch64-darwin> ok      github.com/sigstore/gitsign/internal/attest     0.327s
gitsign-aarch64-darwin> ok      github.com/sigstore/gitsign/internal/cache      0.233s
gitsign-aarch64-darwin> ?       github.com/sigstore/gitsign/internal/commands/attest    [no test files]
gitsign-aarch64-darwin> ok      github.com/sigstore/gitsign/internal/commands/show      0.160s
gitsign-aarch64-darwin> ok      github.com/sigstore/gitsign/internal/config     0.147s
gitsign-aarch64-darwin> --- FAIL: TestVerifyChain (0.02s)
gitsign-aarch64-darwin>     verify_test.go:171: expected x509.UnknownAuthorityError, got x509: OSStatus -26276
gitsign-aarch64-darwin> FAIL
gitsign-aarch64-darwin> FAIL    github.com/sigstore/gitsign/internal/fork/ietf-cms      0.816s
gitsign-aarch64-darwin> FAIL
```

Same test failure on x86_64-darwin:

````
gitsign-x86_64-darwin> running tests
gitsign-x86_64-darwin> ok      github.com/sigstore/gitsign/internal/attest     1.508s
gitsign-x86_64-darwin> ok      github.com/sigstore/gitsign/internal/cache      0.697s
gitsign-x86_64-darwin> ?       github.com/sigstore/gitsign/internal/commands/attest    [no test files]
gitsign-x86_64-darwin> ok      github.com/sigstore/gitsign/internal/commands/show      0.621s
gitsign-x86_64-darwin> ok      github.com/sigstore/gitsign/internal/config     0.495s
gitsign-x86_64-darwin> --- FAIL: TestVerifyChain (0.03s)
gitsign-x86_64-darwin>     verify_test.go:171: expected x509.UnknownAuthorityError, got x509: OSStatus -26276
gitsign-x86_64-darwin> FAIL
gitsign-x86_64-darwin> FAIL    github.com/sigstore/gitsign/internal/fork/ietf-cms      1.255s
gitsign-x86_64-darwin> FAIL
````

LeSuisse commented:
Thanks, I will take a look and/or report upstream.

LeSuisse commented:
I'm not sure this is an upstream issue, this looks similar to #182889

mweinelt commented:
Tests completed on x86-64_darwin on ofBorg. Funky.

LeSuisse commented:
I rebased this just in case.

@developer-guy Would you mind checking if you are able to reproduce the issue on macOS?

developer-guy commented:
Everything seems to be working I've tested on my macOS amd64, and here is the ss from the build:
<img width="1222" alt="Screen Shot 2023-03-12 at 9 29 18 PM" src="https://user-images.githubusercontent.com/16693043/224564888-bc11b85c-a2d2-452f-9754-ea035260a11d.png">

developer-guy reviewed (approved):
LGTM

LeSuisse committed:
gitsign: 0.4.1 -> 0.6.0

https://github.com/sigstore/gitsign/releases/tag/v0.6.0
https://github.com/sigstore/gitsign/releases/tag/v0.5.2

LeSuisse committed:
gitsign: add passthru.tests.version

LeSuisse commented:
I had completely forgotten about this PR :/ 

Anyway I rebased and updated to the freshly released 0.6.0.
